src/sidebar/test/oauth-auth-test.js

```diff
@@ -189,6 +216,32 @@ describe('oauth auth', function () {
           assert.equal(fakeHttp.post.callCount, 1);
         });
     });
+
+    context('when a refresh request fails', function() {
+      beforeEach('make refresh token requests fail', function () {
+        fakeHttp.post = function(url, queryString) {
+          if (queryString.indexOf('refresh_token') !== -1) {
+            return Promise.resolve({status: 500});
+          }
+          return Promise.resolve(successfulFirstAccessTokenPromise);
+        };
+      });
```

Comment:
This kind of thing is what tempts me to wrap things like `$http` in domain-specific interfaces, specific to the domain of `oauth-auth`. Something like a module with `postGrantTokenRequest()` and `postRefreshTokenRequest()` functions that would be simple one-liners calling `$http`. Then the tests for `oauth-auth` could just mock `postGrantTokenRequest()` and/or `postRefreshTokenRequest()` to fail or succeed, rather than having this kind of complexity and being coupled to the contents of refresh request query strings.